heron/stmgr/src/cpp/manager/stmgr-server.cpp

```diff
@@ -162,7 +173,18 @@ sp_string StMgrServer::MakeBackPressureCompIdMetricName(const sp_string& instanc
 }
 
 sp_string StMgrServer::MakeQueueSizeCompIdMetricName(const sp_string& instanceid) {
-  return METRIC_QUEUE_SIZE_TO_INSTANCE_COMPID + instanceid;
+  return CONNECTION_BUFFER_BY_INSTANCEID + instanceid;
+}
+
+void StMgrServer::UpdateQueueMetrics(EventLoop::Status) {
+  for (auto itr = active_instances_.begin(); itr != active_instances_.end(); ++itr) {
+    sp_int32 task_id = itr->second;
+    const sp_string& instance_id = instance_info_[task_id]->instance_->instance_id();
+    sp_int32 bytes = itr->first->getOutstandingBytes();
+    connection_buffer_metric_map_[instance_id]->scope("bytes")->record(bytes);
```

Comment:
hey @huijunwu. What i'd like to have is the average queue length using the last N one second samples. For example If i want to ckeck this every minute, then I need the average over 60 samples (each sample taken every sec). Maybe I didn't understand the code correctly. Are you saying that this accumulates over time  instead of doing the above?